development/scripts/fix-spec.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import yaml
+
+SPEC_PATH = '/client/api/openapi.yaml'
+
+# Load the spec file
+with open(SPEC_PATH, 'r') as file:
+    data = yaml.load(file, Loader=yaml.CLoader)
+
+# Traverse schemas
+if 'components' in data and 'schemas' in data['components']:
+    for name, schema in data['components']['schemas'].items():
+        # Handle failover_strategy
+        if 'failover_strategy' in schema.get('properties', {}):
+            prop = schema['properties']['failover_strategy']
+            if 'default' in prop and prop['default'] is None:
+                print(f"Removing 'default: null' in {name}.failover_strategy")
+                del prop['default']
+                
+        # Handle PowerFeed schema
+        if name == 'PowerFeed' and 'properties' in schema:
+            # Replace nested `type` field in PowerFeed
+            if 'type' in schema['properties']:
+                type_property = schema['properties']['type']
+                if 'properties' in type_property and 'value' in type_property['properties']:
+                    value_property = type_property['properties']['value']
+                    if 'enum' in value_property and set(value_property['enum']) == {'primary', 'redundant'}:
+                        print(f"Replacing complex 'type' field in PowerFeed")
+                        schema['properties']['type'] = {
+                            'type': 'string',
+                            'enum': ['primary', 'redundant'],
+                            'default': 'primary'
+                        }
```

Comment:
For reference:

```yaml
        type:
          type: object
          properties:
            value:
              type: string
              enum:
              - primary
              - redundant
            label:
              type: string
              enum:
              - Primary
              - Redundant
          default:
            value: primary
            label: Primary
```

Reply:
This is an intended asymmetry in the API - on responses from the server, `type` should be a dict with keys `value` and `label`, but on requests to the server, the client should just specify a string corresponding to the `value`. The `PowerFeed` type should only be used on the response side, on the request side see `WritablePowerFeedRequest`, `PatchedWritablePowerFeedRequest`, `BulkWritablePowerFeedRequest`, `PatchedBulkWritablePowerFeedRequest`

Same comment on the other enum fields below.

Reply:
Is this something we need to change here then and/or something that needs to be changed in the specification?
How should go-nautobot handle this?

Reply:
Not sure yet. I currently believe that the generated schema is correct but there may be some nuance I'm missing.